src/components/complex/Proposal/index.jsx

```diff
@@ -239,6 +243,87 @@ const Proposal = ({
     }
   }, [noData, activeChainId])
 
+  const handleMessageFromIframe = (event) => {
+    // Check the origin of the message for security
+    if (event.origin !== window.location.origin) {
+      console.error('Received an untrusted message from:', event.origin)
+      return
+    }
+
+    // Log or handle the received message
+    if (event.data.type === 'linkClick') {
+      if (event.data.id !== id) return
+      setReadMoreContent(event.data.href)
+      setLocation({ current: event.data.href, old: location.current })
+    }
+  }
+
+  const goBack = () => {
+    setReadMoreContent(location.old)
+    setLocation({ current: location.old, old: url })
+  }
+
+  useEffect(() => {
```

Comment:
Are you sure this useEffect doesn't add unnecessary handlers?
It's called without deps, so it's rendered every time.

Reply:
In order to handle ipfs document linked to the ipfs proposal it is necessary to retreive the link clicked and to do that in the safest possible way a script is added to the ipfs proposal with a postMessage (https://github.com/pnetwork-association/dao-v2-ui/blob/3871df33213a6558e03d286b66e34ac607345672/src/components/complex/Proposal/index.jsx#L299) containing the information of the link clicked at each clicked link is generated. This is handled with https://github.com/pnetwork-association/dao-v2-ui/blob/3871df33213a6558e03d286b66e34ac607345672/src/components/complex/Proposal/index.jsx#L267